### PR TITLE
fix: parse S3 target URI when creating empty export folders

### PR DESCRIPTION
**Pull request by Sgitario**
## Summary

When a floorplan query returns no rows, Floorist creates a folder marker via boto3 put_object. That path used AWS_BUCKET verbatim, which breaks for Dataverse Snowpipe credentials where the bucket secret combines bucket and prefix (e.g. dataverse-dev-snowpipe/source-dev-subscriptionwatch/).

Non-empty exports already worked because awswrangler accepts the combined s3:// target. Align the empty-result branch by splitting bucket and key from the same target URI used for parquet writes.

## Solution
- Add `_parse_s3_uri()` to split `s3://` targets into `(bucket, key)`.
- Use the parsed bucket/key when creating the empty folder marker so Snowpipe and simple MinIO bucket configs behave consistently.

## Summary by Sourcery

Align S3 empty-result export behavior with non-empty exports by deriving the bucket and key from the target S3 URI instead of using the configured bucket name verbatim.

Bug Fixes:
- Ensure empty parquet exports correctly split bucket and key from S3 URIs where the configured bucket includes a key prefix (e.g., Snowpipe-style combined bucket and prefix).

Tests:
- Add unit tests for parsing S3 URIs into bucket and key components, including Snowpipe-style buckets with prefixes.
- Add regression tests verifying empty parquet exports use the parsed bucket from the target URI for both prefixed and simple bucket configurations.

### CODE REVIEW

File: src/floorist/floorist.py [hunk 1]
```diff
@@ -96,7 +111,8 @@ def write_parquet(self, data, target, path):
         if len(data) > 0:
             wr.s3.to_parquet(data, target, index=False, compression="gzip", dataset=True, mode="append")
         else:
-            wr._utils.client("s3").put_object(Bucket=self.bucket_name, Body="", Key=path + "/")
+            bucket, key = _parse_s3_uri(target)
+            wr._utils.client("s3").put_object(Bucket=bucket, Body="", Key=f"{key.rstrip('/')}/")
```
Comment (vkrizan) [inline src/floorist/floorist.py:110]: The `self.bucket_name` use should be preserved.

File: src/floorist/floorist.py [hunk 2]
```diff
@@ -96,7 +111,8 @@ def write_parquet(self, data, target, path):
         if len(data) > 0:
             wr.s3.to_parquet(data, target, index=False, compression="gzip", dataset=True, mode="append")
         else:
-            wr._utils.client("s3").put_object(Bucket=self.bucket_name, Body="", Key=path + "/")
+            bucket, key = _parse_s3_uri(target)
```
Comment (vkrizan) [inline src/floorist/floorist.py]: There is a `path` passed just for this `if` block. Could this improved to pass whatever it's necessary to avoid parsing raw strings?

File: src/floorist/floorist.py (hunk 1 above)
Comment (Sgitario) [reply src/floorist/floorist.py:110]: it makes sense to me. Changed.

File: src/floorist/floorist.py (hunk 2 above)
Comment (Sgitario) [reply src/floorist/floorist.py]: I simplified these changes.

**Review by vkrizan — APPROVED** [review]

**Review by patchkez — APPROVED** [review]

File: tests/test_floorist_standalone.py [hunk 3]
```diff
@@ -400,3 +401,80 @@ def track_query(*args, **kwargs):
...
+            Body="",
+            Key="empty/year_created=2026/month_created=6/day_created=3/",
+        )
+
+    @patch("floorist.floorist.wr.s3.to_parquet")
+    @patch("floorist.floorist.wr._utils.client")
+    def test_nonempty_export_still_uses_awswrangler(self, mock_client_fn, mock_to_parquet):
+        mock_s3 = Mock()
+        mock_client_fn.return_value = mock_s3
+        client = self._s3_client("dataverse-dev-snowpipe/source-dev-subscriptionwatch/")
```
Comment (vkrizan) [inline tests/test_floorist_standalone.py]: Do you want to keep these names in the tests?

Comment (Sgitario) [reply tests/test_floorist_standalone.py]: +1 just updated the tests to not use any swatch specific names
